File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int>sp(nums.begin(),nums.end());
        int maxLen=0;
        for(auto num:nums){
            if(sp.find(num-1)==sp.end()){
                int curr_num=num;
                int curr_len=1;
                while(sp.find(curr_num+1)!= sp.end()){
                    curr_num+=1;
                    curr_len+=1;
                }
                maxLen = max(maxLen, curr_len);
            }
        }
        return maxLen;
    }
};

File: Data Structures & Algorithms/subarray-product-less-than-k/submission-0.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
        if k<=1:return 0
        l=0
        r=0
        prod=1
        count=0

        while r<len(nums):
            prod*=nums[r]

            while prod>=k:
                prod/=nums[l]
                l+=1
            
            count+=(r-l+1)
            r+=1
        return count